src/_isCyclic.ts

```diff
@@ -0,0 +1,53 @@
+const MAX_DEPTH = 1000;
+
+/**
+ * Checks if the given value has any cyclic references.
+ *
+ * @param value - The value to check for cyclic references.
+ * @returns boolean - Returns true if the value contains cyclic references, otherwise false.
+ * @example
+ * ```ts
+ * isCyclic(value)
+ * ```
+ */
+export function isCyclic(value: unknown): boolean {
+  const seenValues = new WeakSet(); // use to keep track of which objects have been seen.
+  let depth = 0;
+
+  function hasCycle(input: unknown) {
+    depth++;
+
+    // If max depth is exceeded, return false
```

Comment:
it won't work as intended. the exit will not be from the root call, in whose scope the variable is defined, but from one of the recursive calls. Thus, we will simply stop one of the iterations without stopping the function as a whole, and continue to count again in order to skip another one after 1000 iterations.

Demo: https://codesandbox.io/s/xenodochial-gagarin-hynlcd?file=/src/index.ts

Reply:
It can be solved with try/catch and custom exception:

```js
const maxDepthReached = Symbol("maxDepthReached"); // on top level

...

    if (depth > MAX_DEPTH) {
      throw maxDepthReached;
    }

...

  try {
    return hasCycle(value);
  } catch (exception) {
    if (exception === maxDepthReached) {
      return false;
    }
    throw exception;
  }
```

demo: https://codesandbox.io/s/pedantic-phoebe-5xqcxc?file=/src/index.ts

Reply:
Instead of a symbol, you can use another way to define your exception, for example, extend the error class and check instancesof. The variant with the symbol looks simpler.

Reply:
Removed depth checking